### DISCUSSION TITLE
Global State

### DISCUSSION THREAD

**Discussion by TimUntersberger** [ivanceras/sauron#45] [General]
Hey,

I am currently looking for a rust based web framework to try and right now sauron is my favorite, but I am wondering whether you currently support global state or are at least planning on adding support?

**Comment by ivanceras** [comment]
Hello @TimUntersberger I'm not quite sure what you mean about global state support. The thing that comes into my mine when you say global state is using crates such as `once_cell`, `lazy_static`, or even the combination of `Arc` and `Mutex` from the standard library would do the job. Can you give me a use case where global state is used?

**Reply by TimUntersberger** [reply]
Yeah I forgot that you can just use something like lazy_static for this purpose. Thank you.

**Reply by ivanceras** [reply]
I believe [`once_cell`](https://docs.rs/once_cell/)  is a better crate for what you are trying to do. It has all the capability of what `lazy_static` is trying to do, and in the process of being integrated into the [std library](https://doc.rust-lang.org/stable/std/lazy/struct.OnceCell.html)
